{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"name":"SKIOME_code.ipynb","provenance":[],"collapsed_sections":[]},"kernelspec":{"name":"python3","display_name":"Python 3"},"language_info":{"name":"python"}},"cells":[{"cell_type":"markdown","source":["# SKIOME Datasets and Metadata Retrieval - Pipeline\n","\n","### Download SRAdb\n","Install SRAdb package if not present\n","\n","Use SRAdb package to download the SRAdb database containing all the metadata associated with publicly available datasets.\n","\n","### Download the database (>36 Gb):"],"metadata":{"id":"LEZ6cJndczS3"}},{"cell_type":"code","source":["library(SRAdb)\n","\n","# Download SRAdb:\n","sqlfile <- getSRAdbFile()"],"metadata":{"id":"Vo4Up3QpdAcq"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["## PART 1 - Search studies of interest with SRAdb and create a dataframe with all the metadata retrived with SRAdb and EDirect\n","\n","### SRAdb querying and metadata download:\n","Query of the SRAdb database with a full-text search to recover a list of studies of interest with their associated metadata."],"metadata":{"id":"xPNLIXgBdFrZ"}},{"cell_type":"code","source":["library(SRAdb)\n","\n","# Connect to the db:\n","sra_con <- dbConnect(SQLite(),sqlfile)\n","\n","SRA_dataframe <- getSRA( search_terms ='human skin microbiome OR human skin microbiota OR human skin metagenome', \n","                    out_types = c(\"col_desc\", \"experiment\", \"fastq\", \"metaInfo\", \"run\", \"sample\", \"sra\", \"sra_ft\",\n","                                  \"sra_ft_content\", \"sra_ft_segdir\", \"sra_ft_segments\", \"study\", \"submission\"), sra_con )\n","dim(SRA_dataframe)"],"metadata":{"id":"ncCYUsawdPGk"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["Extract from the dataframe created with SRAdb the list of studies The recovered list can be used to download further information on the run with EDirect tool"],"metadata":{"id":"nS_PAE61dUEf"}},{"cell_type":"code","source":["library(readr)\n","auto_list <- unique(SRA_dataframe$study)\n","auto_list <- as.data.frame(studies_list)\n","write_tsv(auto_list, col_names = F, path = \"EDirect_metadata/SRA_studies_list.txt\")\n","# Note: it is advisable to store this file in a sub-directory where to run EDirect and store all the metadata files (e.g. \"EDirect_metadata\" directory)"],"metadata":{"id":"S14f_3LjdTiH"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["Use the generated file to download run_info file with EDirect.\n","\n","### Download metadata with EDirect (from a Unix terminal window)\n","\n","See https://www.ncbi.nlm.nih.gov/books/NBK179288/ for EDirect installation instructions."],"metadata":{"id":"ts37l91sdY1w"}},{"cell_type":"code","source":["sample_list=$(cat  SRA_studies_list.txt)\n","for a in $sample_list\n","   do\n","   esearch -db sra -query \"$a\" | efetch -format runinfo > \"$a\"_runinfo # download metadata for the runs\n","done"],"metadata":{"id":"VY-UvH9SdjT3"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["### Import metadata obtained from EDirect tool into a single dataframe"],"metadata":{"id":"MRRvWw0Odmyz"}},{"cell_type":"code","source":["# set the directory to where the EDirect runifo files are located\n","setwd(\"FULL_PATH/EDirect_metadata\")\n","\n","# Create a list with the names of the EDirect files:\n","# convert to vector of characters\n","auto_list <- as.character(auto_list$studies_list)\n","# append the _runinfo to each character to create a vector with the names of the file you need to import\n","auto_list <- paste(auto_list, \"_runinfo\", sep = \"\")\n","\n","# create a empty dataframe:\n","auto_edirect <- data.frame(matrix(ncol = 47, nrow = 0))\n","# import one file as a temporary variable\n","tmp <- read.csv(auto_list[1])\n","# use the imported file to name the dataframe columns\n","colnames(auto_edirect) <- colnames(tmp)\n","# remove the temporary file\n","rm(tmp)\n","\n","# import all the datasets:\n","for (k in 1:length(auto_list)){\n","  tmp <- read.csv(auto_list[k])\n","  # remove redundant headers\n","  for (n in nrow(tmp):1){\n","    if (tmp[n, 1]==\"Run\"){\n","      tmp <- tmp[-c(n),]\n","    }\n","  }\n","  # merge datasets\n","  auto_edirect <- rbind(auto_edirect, tmp)\n","  rownames(auto_edirect) <- 1:nrow(auto_edirect)\n","}\n","\n","setwd(\"FULL_PATH_TO_WORKING_DIRECTORY\")"],"metadata":{"id":"3eK5PtWkdp3a"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["### Join SRAdb and E-Direct information in a single dataframe:"],"metadata":{"id":"9lTXjqlBds4Z"}},{"cell_type":"code","source":["library(dplyr)\n","colnames(SRA_dataframe)[2] <- \"Run\"\n","auto_df <- merge(SRA_dataframe, auto_edirect, by = c(\"Run\"), all.x = T, all.y = T)"],"metadata":{"id":"0x3biGhHdv73"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["### Add release date column:"],"metadata":{"id":"YGX2SbWQd0aV"}},{"cell_type":"code","source":["release <- auto_df$ReleaseDate\n","for (w in 1:length(release)){\n","  release[w] <- substr(release[w], 1, 4)\n","}\n","auto_df <- cbind(auto_df, release)\n","colnames(auto_df)[117] <- \"Year_of_release\""],"metadata":{"id":"ItNYEzP-d575"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["### Reorganize the columns:"],"metadata":{"id":"c1RN8buZd9Ri"}},{"cell_type":"code","source":["curated_df <- auto_df %>% select(Run, Year_of_release, ReleaseDate, LoadDate, Submission, spots.y, bases.y, spots_with_mates, avgLength, size_MB, download_path, Experiment, LibraryStrategy, LibraryLayout, InsertSize, Platform, Model, experiment_title, design_description, library_name, experiment_attribute, Sample, BioSample, TaxID, ScientificName, Sex, Body_Site, sample_alias, description, sample_attribute, CenterName, SRAStudy, BioProject, center_project_name, sradb_updated, study_description, study_abstract)\n","\n","# rename columns\n","colnames(curated_df) <- c(\"Run\", \"Year_of_release\", \"Release_Date\", \"Load_Date\", \"Submission\", \"Spots\", \"Bases\", \"Spots_with_mates\", \"AvgLength\", \"Size_MB\", \"Download_path\", \"Experiment\", \"Library_Strategy\", \"Library_Layout\", \"Insert_Size\", \"Platform\", \"Model\", \"Experiment_title\", \"Design_description\", \"Library_name\", \"Experiment_attribute\", \"Sample_ID\", \"BioSample\", \"TaxID\", \"Scientific_Name\", \"Sex\", \"Body_Site\", \"Sample_alias\", \"Description\", \"Sample_attribute\", \"Center_Name\", \"Study_ID\", \"BioProject\", \"Center_project_name\", \"Sradb_updated\", \"Study_description\", \"Study_abstract\")"],"metadata":{"id":"hW6z-zkMeBXK"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["# PART 2 - Search studies of interest both manually and with SRAdb and create a single dataframe with all the retrieved studies and the associated metadata retrived with SRAdb, EDirect and manually.\n","\n","For the automatic search of SRAdb we will use the same dataframe created in part 1\n","\n","Here we include the manual dataframe with his metadata\n","\n","### Import the manual dataframe:"],"metadata":{"id":"ydvS5eEteDnB"}},{"cell_type":"code","source":["manual_df <- read.csv(\"Human Skin Datasets.csv\")"],"metadata":{"id":"WViVD6NweJOB"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["### Use SRAdb to specifically search for metadata associated with the manually retrieved list of studies\n","\n","To this end we use as query the study IDs:"],"metadata":{"id":"J91orvmGeLOw"}},{"cell_type":"code","source":["library(SRAdb)\n","\n","# Connect to the db:\n","sra_con <- dbConnect(SQLite(),sqlfile)\n","\n","SRA_manual <- getSRA( search_terms ='ERP126714 OR SRP144464 OR SRP091029 OR SRP185615 OR DRP003136 OR SRP071579 OR SRP067274 OR SRP062726 OR\n","                DRP000933OR DRP004449 OR SRP091027 OR ERP017003 OR ERP019566 OR ERP024081 OR ERP016977 OR ERP014521 OR SRP255681 OR \n","                SRP090974 OR ERP104625 OR SRP135853 OR ERP109379 OR ERP005296 OR ERP013063 OR SRP262012 OR SRP276976 OR SRP22682 OR \n","                SRP285536 OR ERP014863 OR ERP107880 OR SRP109593 OR SRP247381 OR SRP063707 OR SRP187334 OR SRP213896 OR ERP013768 OR \n","                SRP252817 OR SRP276001 OR ERP109520 OR SRP034607 OR SRP185750 OR SRP279904 OR SRP109057 OR SRP293771 OR SRP067489 OR \n","                ERP114401 OR SRP276479 OR SRP267292 OR SRP167680 OR ERP112507 OR ERP001059 OR SRP230017 OR SRP275714 OR SRP188413 OR \n","                SRP050070 OR SRP186628 OR ERP105323 OR SRP057641 OR ERP016529 OR ERP022958 OR SRP113602 OR ERP016629 OR SRP214545 OR \n","                ERP018577 OR ERP005182 OR SRP051059 OR SRP133226 OR ERP016280 OR SRP002480', \n","                out_types = c(\"col_desc\", \"experiment\", \"fastq\", \"metaInfo\", \"run\", \"sample\", \"sra\", \"sra_ft\",\n","                                  \"sra_ft_content\", \"sra_ft_segdir\", \"sra_ft_segments\", \"study\", \"submission\"), sra_con )\n","dim(SRA_manual)"],"metadata":{"id":"dMtjZsq_eRWu"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["Note: Not all the manually found studies are found by SRAdb\n","\n","### Merge the manual SRAdb dataset and the automatic SRAdb dataset and remove redundant rows (duplicated samples)"],"metadata":{"id":"gyLW5Jh-eSYQ"}},{"cell_type":"code","source":["unique(SRA_dataframe$study)\n","unique(SRA_manual$study)\n","setdiff(SRA_manual$study, SRA_dataframe$study)\n","setdiff(SRA_dataframe$study, SRA_manual$study)\n","\n","# bind the \"manual\" dataset with the \"sradb search\" dataset\n","merged_df <- rbind(SRA_dataframe, SRA_manual)\n","unique(merged_df$study)\n","\n","# Remove duplicate rows based on certain columns (variables):\n","# Remove duplicated rows based on Sepal.Length\n","merged_df <- merged_df %>% distinct(run, .keep_all = TRUE)\n","length(unique(merged_df$study))"],"metadata":{"id":"GNKvtWI3eWnJ"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["Extract from the dataframes created with SRAdb automatic search and the manual search the list of studies The recovered list can be used to download further information on the run with EDirect tool"],"metadata":{"id":"CoTQNpv3eXOq"}},{"cell_type":"code","source":["library(readr)\n","auto_list <- unique(SRA_dataframe$study)\n","auto_list <- as.data.frame(auto_list)\n","manual_list <- unique(manual_df$Study_alias)\n","manual_list <- as.data.frame(manual_list)\n","colnames(auto_list)[1] <- \"study\"\n","colnames(manual_list)[1] <- \"study\"\n","edirect_list <- rbind(auto_list, manual_list)\n","# remove redundant rows\n","edirect_list <- edirect_list %>% distinct(study, .keep_all = TRUE)\n","# create a list with the study_ID of the samples you still haven't downloaded from EDirect\n","# you will use this list to perform a new EDirect search:\n","list_for_edirect <- setdiff(manual_list$study, auto_list$study)\n","list_for_edirect <- as.data.frame(list_for_edirect)\n","write_tsv(list_for_edirect, col_names = F, path = \"EDirect_metadata/list_for_edirect.txt\")\n","# Note: it is advisable to store this file in a sub-directory where to run EDirect and store all the metadata files"],"metadata":{"id":"KvMmgJrWeZro"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["## Download metadata with EDirect (from a Unix terminal window)"],"metadata":{"id":"xx6gTuhbebwk"}},{"cell_type":"code","source":["sample_list=$(cat  list_for_edirect.txt)\n","for a in $sample_list\n","   do\n","   esearch -db sra -query \"$a\" | efetch -format runinfo > \"$a\"_runinfo # download metadata for the runs\n","done"],"metadata":{"id":"ws5-iBfkef3d"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["### EDirect metadata import for the manual dataset list\n","Import metadata obtained from EDirect tool into a sigle dataframe"],"metadata":{"id":"n0ZIBoZJem-B"}},{"cell_type":"code","source":["# set the directory to where the EDirect runifo files are located\n","setwd(\"/FULL_PATH/EDirect_metadata\")\n","\n","# Create another list with the names of the EDirect files:\n","# convert to vector of characters\n","edirect_list <- as.character(edirect_list$study)\n","# append the _runinfo to each character to create a vector with the names of the file you need to import\n","edirect_list <- paste(edirect_list, \"_runinfo\", sep = \"\")\n","\n","# create a empty dataframe:\n","edirect_df <- data.frame(matrix(ncol = 47, nrow = 0))\n","# import one file in a temporary file\n","tmp <- read.csv(edirect_list[1])\n","# use the imported file to name the dataframe columns\n","colnames(edirect_df) <- colnames(tmp)\n","# remove the temporary file\n","rm(tmp)\n","\n","# import all the datasets:\n","for (k in 1:length(edirect_list)){\n","  tmp <- read.csv(edirect_list[k])\n","  # remove redundant headers\n","  for (n in nrow(tmp):1){\n","    if (tmp[n, 1]==\"Run\"){\n","      tmp <- tmp[-c(n),]\n","    }\n","  }\n","  # merge datasets\n","  edirect_df <- rbind(edirect_df, tmp)\n","  rownames(edirect_df) <- 1:nrow(edirect_df)\n","}\n","\n","setwd(\"FULL_PATH_TO_WORKING_DIRECTORY\")"],"metadata":{"id":"-BVi2vHmeq-N"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["### Merging the dataframes:"],"metadata":{"id":"Ms8Mb1gjetml"}},{"cell_type":"code","source":["library(dplyr)\n","\n","# join SRAdb and E-Direct information in a single dataset:\n","colnames(merged_df)[2] <- \"Run\"\n","merged_df_2 <- merge(merged_df, edirect_df, by = c(\"Run\"), all.x = T, all.y = T)\n","# merge sparse study_id columns:\n","merged_df_2 <- merged_df_2 %>%\n","  mutate(study = coalesce(study,SRAStudy))\n","# include manual retrieved study-specific information:\n","merged_df_2 <- left_join(merged_df_2, manual_df, by = c(\"study\"))"],"metadata":{"id":"VR2R7WGjex1P"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["### Add release date column:"],"metadata":{"id":"ww_jeVU6ewyN"}},{"cell_type":"code","source":["release <- merged_df_2$ReleaseDate\n","for (w in 1:length(release)){\n","  release[w] <- substr(release[w], 1, 4)\n","}\n","merged_df_2 <- cbind(merged_df_2, release)\n","colnames(merged_df_2)[146] <- \"Year_of_release\""],"metadata":{"id":"sWtA2Rw1e7IX"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["### Reorganize the columns:"],"metadata":{"id":"YOhjSN3Te-nb"}},{"cell_type":"code","source":["curated_df_2 <- merged_df_2 %>% select(Run, Year_of_release, ReleaseDate, LoadDate, Submission, spots.y, bases.y, spots_with_mates, avgLength, size_MB, download_path, Experiment, LibraryStrategy, LibraryLayout, InsertSize, Platform, Model, experiment_title, design_description, library_name, experiment_attribute, Sample, BioSample, TaxID, ScientificName, Sex, Body_Site, sample_alias, description, sample_attribute, CenterName, SRAStudy, BioProject, center_project_name, sradb_updated, collection_method, vRegion, clustering_used, rSequences.OTUs.ASVs., taxon_database.used_in_the_study., taxon_database_version, coocurrence_study, disease.condition, Location, MGnify.analysis, DOI, Year_Of_Publication, Journal, WOS_Research_Areas, WOS_Categories, Scopus_Research_Subject_1, Scopus_Research_Subject_2, Scopus_Research_Subject_3, Scopus_Research_Subject_4, Scopus_Research_Subject_5, Scopus_Research_Subject_6, Medicine_journal, study_description, study_abstract, Notes., Manual_Validation)\n","\n","# rename columns:\n","colnames(curated_df_2) <- c(\"Run\", \"Year_of_release\", \"Release_Date\", \"Load_Date\", \"Submission\", \"Spots\", \"Bases\", \"Spots_with_mates\", \"AvgLength\", \"Size_MB\", \"Download_path\", \"Experiment\", \"Library_Strategy\", \"Library_Layout\", \"Insert_Size\", \"Platform\", \"Model\", \"Experiment_title\", \"Design_description\", \"Library_name\", \"Experiment_attribute\", \"Sample_ID\", \"BioSample\", \"TaxID\", \"Scientific_Name\", \"Sex\", \"Body_Site\", \"Sample_alias\", \"Description\", \"Sample_attribute\", \"Center_Name\", \"Study_ID\", \"BioProject\", \"Center_project_name\", \"Sradb_updated\", \"Collection_method\", \"Region_16S\", \"Clustering_used\", \"nSequences_OTUs_ASVs\", \"Taxon_database_used_in_the_study\", \"Taxon_database_version\", \"Coocurrence_study\", \"Disease_condition\", \"Location\", \"MGnify_analysis\", \"DOI\", \"Year_Of_Publication\", \"Journal\", \"WOS_Research_Areas\", \"WOS_Categories\", \"Scopus_Research_Subject_1\", \"Scopus_Research_Subject_2\", \"Scopus_Research_Subject_3\", \"Scopus_Research_Subject_4\", \"Scopus_Research_Subject_5\", \"Scopus_Research_Subject_6\", \"Medicine_journal\", \"Study_description\", \"Study_abstract\", \"Notes\", \"Manual_Validation\")"],"metadata":{"id":"OORFXeiWfERi"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["### Subset only manually retrieved studies:"],"metadata":{"id":"xj_pz1tefG4j"}},{"cell_type":"code","source":["manual_dataframe <- filter(curated_df_2, Manual_Validation == \"Manually_Validated\")\n","length(unique(manual_dataframe$Study_ID))"],"metadata":{"id":"0bryhst5fLq8"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["### Remove rows and correct errors:"],"metadata":{"id":"p7KocX6yfNbz"}},{"cell_type":"code","source":["# correct two errors in the library strategy metadata:\n","# correction applied based on the information on the published article\n","library(dplyr)\n","manual_dataframe <- manual_dataframe %>% mutate(Library_Strategy = ifelse(Study_ID == \"ERP126714\" & Library_Strategy == \"RNA-Seq\", \"AMPLICON\", Library_Strategy))\n","manual_dataframe <- manual_dataframe %>% mutate(Library_Strategy = ifelse(Study_ID == \"ERP024081\" & Library_Strategy == \"WGS\", \"AMPLICON\", Library_Strategy))\n","# remove WGS rows:\n","manual_dataframe <- manual_dataframe[manual_dataframe$Library_Strategy != \"WGS\",]"],"metadata":{"id":"5uKB11CcfQf7"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["### Collapse the Scopus research subjects into a single column:"],"metadata":{"id":"qHSBQwrb_EQh"}},{"cell_type":"code","source":["manual_dataframe$Scopus_Research_Subject <- paste(manual_dataframe$Scopus_Research_Subject_1, manual_dataframe$Scopus_Research_Subject_2, manual_dataframe$Scopus_Research_Subject_3, manual_dataframe$Scopus_Research_Subject_4, manual_dataframe$Scopus_Research_Subject_5, manual_dataframe$Scopus_Research_Subject_6, sep = \"; \")\n","\n","curated_df_2$Scopus_Research_Subject <- paste(curated_df_2$Scopus_Research_Subject_1, curated_df_2$Scopus_Research_Subject_2, curated_df_2$Scopus_Research_Subject_3, curated_df_2$Scopus_Research_Subject_4, curated_df_2$Scopus_Research_Subject_5, curated_df_2$Scopus_Research_Subject_6, sep = \"; \")"],"metadata":{"id":"F0qal7kc_IxK"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["### Export the dataframes:"],"metadata":{"id":"7zGCiAsTfSDg"}},{"cell_type":"code","source":["write.csv(curated_df, \"Dataframe_1_Automatic.csv\", row.names = FALSE)\n","write.csv(curated_df_2, \"Dataframe_2.csv\", row.names = FALSE)\n","write.csv(manual_dataframe, \"Dataframe_3_Manual.csv\", row.names = FALSE)"],"metadata":{"id":"O5rurtt2fYQD"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["## Data Frames exploration:"],"metadata":{"id":"OnbvPPMKfbIT"}},{"cell_type":"code","source":["# describe the datasets\n","dim(curated_df)\n","length(unique(curated_df$Study_ID))\n","\n","dim(curated_df_2)\n","length(unique(curated_df_2$Study_ID))\n","\n","dim(manual_dataframe)\n","length(unique(manual_dataframe$Study_ID))\n","\n","# compare manual vs automatic search\n","length(setdiff(unique(curated_df$Study_ID), unique(manual_dataframe$Study_ID)))\n","length(setdiff(unique(manual_dataframe$Study_ID), unique(curated_df$Study_ID)))\n","\n","# Year\n","unique(curated_df$Year_of_release)\n","unique(manual_dataframe$Year_of_release)\n","unique(manual_dataframe$Year_Of_Publication)\n","\n","length(which(curated_df$Year_of_release == \"2015\"))\n","length(which(curated_df_2$Year_of_release == \"2017\"))\n","length(which(manual_dataframe$Year_of_release == \"2017\"))\n","length(which(manual_dataframe$Year_of_release == \"2013\"))\n","length(which(manual_df$Year_Of_Publication == \"2019\"))\n","\n","length(which(curated_df$Year_of_release == \"2008\"))\n","length(which(curated_df$Year_of_release == \"2009\"))\n","length(which(curated_df$Year_of_release == \"2010\"))\n","length(which(curated_df$Year_of_release == \"2011\"))\n","\n","# collection method:\n","unique(manual_dataframe$Collection_method)\n","length(which(manual_df$collection_method == \"swab\"))\n","length(which(manual_df$collection_method == \"biopsy\"))\n","length(which(manual_df$collection_method == \"scrubs buffer washes\"))\n","length(which(manual_df$collection_method == \"swab and tape-strip\"))\n","length(which(manual_df$collection_method == \"swab and biopsy\"))\n","length(which(manual_df$collection_method == \"swab, pore strip, glue strip\"))\n","\n","length(which(manual_dataframe$Collection_method == \"swab\"))\n","length(which(manual_dataframe$Collection_method == \"biopsy\"))\n","length(which(manual_dataframe$Collection_method == \"scrubs buffer washes\"))\n","length(which(manual_dataframe$Collection_method == \"swab and tape-strip\"))\n","length(which(manual_dataframe$Collection_method == \"swab and biopsy\"))\n","length(which(manual_dataframe$Collection_method == \"swab, pore strip, glue strip\"))\n","\n","# 16S region:\n","# studies:\n","unique(manual_df$vRegion)\n","length(which(manual_df$vRegion == \"Many (V1-V3 & V1-V8)\"))\n","length(which(manual_df$vRegion == \"V1-V2\"))\n","length(which(manual_df$vRegion == \"V1-V2 and v4\"))\n","length(which(manual_df$vRegion == \"V1-V3\"))\n","length(which(manual_df$vRegion == \"V1-V4\"))\n","length(which(manual_df$vRegion == \"V2-V3\"))\n","length(which(manual_df$vRegion == \"V3-V4\"))\n","length(which(manual_df$vRegion == \"V3-V5\"))\n","length(which(manual_df$vRegion == \"V4\"))\n","length(which(manual_df$vRegion == \"V4-V5\"))\n","# samples:\n","unique(manual_dataframe$Region_16S)\n","length(which(manual_dataframe$Region_16S == \"Many (V1-V3 & V1-V8)\"))\n","length(which(manual_dataframe$Region_16S == \"V1-V2\"))\n","length(which(manual_dataframe$Region_16S == \"V1-V2 and v4\"))\n","length(which(manual_dataframe$Region_16S == \"V1-V3\"))\n","length(which(manual_dataframe$Region_16S == \"V1-V4\"))\n","length(which(manual_dataframe$Region_16S == \"V2-V3\"))\n","length(which(manual_dataframe$Region_16S == \"V3-V4\"))\n","length(which(manual_dataframe$Region_16S == \"V3-V5\"))\n","length(which(manual_dataframe$Region_16S == \"V4\"))\n","length(which(manual_dataframe$Region_16S == \"V4-V5\"))\n","\n","# seq platform:\n","# manual dataframe\n","unique(manual_dataframe$Platform)\n","length(which(manual_dataframe$Platform == \"LS454\"))\n","length(which(manual_dataframe$Platform == \"ILLUMINA\"))\n","\n","unique(manual_dataframe$Model)\n","length(which(manual_dataframe$Model == \"Illumina MiSeq\"))\n","length(which(manual_dataframe$Model == \"Illumina MiniSeq\"))\n","length(which(manual_dataframe$Model == \"Illumina HiSeq 2000\"))\n","length(which(manual_dataframe$Model == \"Illumina HiSeq 2500\"))\n","length(which(manual_dataframe$Model == \"Illumina NovaSeq 6000\"))\n","length(which(manual_dataframe$Model == \"454 GS FLX Titanium\"))\n","length(which(manual_dataframe$Model == \"454 GS FLX+\"))\n","length(which(manual_dataframe$Model == \"454 GS FLX\"))\n","length(which(manual_dataframe$Model == \"454 GS\"))\n","length(which(manual_dataframe$Model == \"unspecified\"))\n","\n","# manual + automatic dataframe\n","unique(curated_df_2$Platform)\n","length(which(curated_df_2$Platform == \"LS454\"))\n","length(which(curated_df_2$Platform == \"ILLUMINA\"))\n","length(which(curated_df_2$Platform == \"ION_TORRENT\"))\n","length(which(curated_df_2$Platform == \"CAPILLARY\"))\n","\n","unique(curated_df_2$Model)\n","length(which(curated_df_2$Model == \"Illumina MiSeq\"))\n","length(which(curated_df_2$Model == \"Illumina MiniSeq\"))\n","length(which(curated_df_2$Model == \"Illumina HiSeq 2000\"))\n","length(which(curated_df_2$Model == \"Illumina HiSeq 2500\"))\n","length(which(curated_df_2$Model == \"Illumina NovaSeq 6000\"))\n","length(which(curated_df_2$Model == \"454 GS FLX Titanium\"))\n","length(which(curated_df_2$Model == \"454 GS FLX+\"))\n","length(which(curated_df_2$Model == \"454 GS FLX\"))\n","length(which(curated_df_2$Model == \"454 GS\"))\n","length(which(curated_df_2$Model == \"unspecified\"))\n","\n","# Clustering used:\n","unique(manual_df$clustering_used)\n","length(which(manual_df$clustering_used == \"OTUs\"))\n","length(which(manual_df$clustering_used == \"ASVs\"))\n","length(which(manual_df$clustering_used == \"RSVs\"))\n","sum(is.na(manual_df$clustering_used))\n","\n","# Taxonomic database used:\n","unique(manual_df$taxon_database.used_in_the_study.)\n","length(which(manual_df$taxon_database.used_in_the_study. == \"Greengenes\"))\n","length(which(manual_df$taxon_database.used_in_the_study. == \"SILVA\"))\n","length(which(manual_df$taxon_database.used_in_the_study. == \"RDP\"))\n","length(which(manual_df$taxon_database.used_in_the_study. == \"NCBI\"))\n","length(which(manual_df$taxon_database.used_in_the_study. == \"EzTaxon-e\"))\n","length(which(manual_df$taxon_database.used_in_the_study. == \"HOMD and Greengenes\"))\n","sum(is.na(manual_df$taxon_database.used_in_the_study.))\n","\n","# Co-occurrence analysis:\n","length(which(manual_df$coocurrence_study == \"Yes\"))\n","\n","# Disease/condition:\n","unique(manual_df$disease.condition)\n","length(which(manual_df$disease.condition == \"Healthy\"))\n","\n","length(which(manual_df$disease.condition == \"Atopic Dermatitis\"))\n","length(which(manual_df$disease.condition == \"Healthy & Atopic Dermatitis\"))\n","length(which(manual_df$disease.condition == \"Atopic Dermatitis and Psoriasis\"))\n","length(which(manual_df$disease.condition == \"Psoriasis\"))\n","length(which(manual_df$disease.condition == \"Parapsoriasis\"))\n","\n","length(which(manual_df$disease.condition == \"Acne\"))\n","length(which(manual_df$disease.condition == \"Dandruff\"))\n","length(which(manual_df$disease.condition == \"Leprosy\"))\n","length(which(manual_df$disease.condition == \"Hidradenitis Suppurativa\"))\n","length(which(manual_df$disease.condition == \"Squamous Cell Carcinoma\"))\n","length(which(manual_df$disease.condition == \"Filaggrin-deficient human skin\"))\n","length(which(manual_df$disease.condition == \"Autoimmune bullous disease\"))\n","length(which(manual_df$disease.condition == \"Dystrophic epidermolysis bullosa\"))\n","length(which(manual_df$disease.condition == \"Vitiligo\"))\n","\n","# skin injuries:\n","length(which(manual_df$disease.condition == \"Burn Injury\"))\n","length(which(manual_df$disease.condition == \"Diabetic Wound\"))\n","length(which(manual_df$disease.condition == \"Damaged skin\"))\n","length(which(manual_df$disease.condition == \"Wound\"))\n","length(which(manual_df$disease.condition == \"Chronic wounds\"))\n","length(which(manual_df$disease.condition == \"Open fractures\"))\n","\n","# allegic traits and atopic individuals:\n","length(which(manual_df$disease.condition == \"Allergic traits\"))\n","length(which(manual_df$disease.condition == \"Atopic individuals\"))\n","\n","# skin pathogenic infections (bacterial, fungal and parasitic infection)\n","length(which(manual_df$disease.condition == \"Tinea Pedis\"))\n","length(which(manual_df$disease.condition == \"Cutaneous leishmaniasis lesions\"))\n","length(which(manual_df$disease.condition == \"Buruli ulcer skin lesions\"))\n","\n","# other conditions:\n","length(which(manual_df$disease.condition == \"Low birthweight\"))\n","length(which(manual_df$disease.condition == \"Obesity\"))\n","\n","# Location:\n","unique(manual_df$Location)\n","length(which(manual_df$Location == \"China\"))\n","length(which(manual_df$Location == \"USA\"))\n","\n","# Europe:\n","length(which(manual_df$Location == \"Ireland\"))\n","length(which(manual_df$Location == \"Finland\"))\n","length(which(manual_df$Location == \"Denmark\"))\n","length(which(manual_df$Location == \"Belgium\"))\n","length(which(manual_df$Location == \"Netherlands\"))\n","length(which(manual_df$Location == \"Germany\"))\n","length(which(manual_df$Location == \"France\"))\n","length(which(manual_df$Location == \"Austria\"))\n","length(which(manual_df$Location == \"UK\"))\n","length(which(manual_df$Location == \"Europe\"))\n","\n","# Other:\n","length(which(manual_df$Location == \"Canada and UK\"))\n","length(which(manual_df$Location == \"Canada\"))\n","length(which(manual_df$Location == \"Japan\"))\n","length(which(manual_df$Location == \"Australia\"))\n","length(which(manual_df$Location == \"Korea\"))\n","length(which(manual_df$Location == \"India\"))\n","length(which(manual_df$Location == \"Brazil\"))\n","length(which(manual_df$Location == \"Israel\"))\n","length(which(manual_df$Location == \"Egypt\"))\n","length(which(manual_df$Location == \"Benin (Africa)\"))\n","\n","# Taxon ID and scientific name\n","unique(manual_dataframe$TaxID)\n","unique(manual_dataframe$Scientific_Name)\n","\n","length(unique(curated_df_2$TaxID))\n","unique(curated_df_2$Scientific_Name)\n","\n","# Spots: dataframe1\n","sum(is.na(curated_df$Spots))\n","min(as.numeric(curated_df$Spots))\n","length(which(curated_df$Spots == 0))\n","max(as.numeric(curated_df$Spots))\n","median(as.numeric(curated_df$Spots))\n","tmp <- as.numeric(curated_df$Spots)\n","mean(tmp[tmp != 0])\n","median(tmp[tmp != 0])\n","\n","# spots: dataframe2\n","median(as.numeric(curated_df_2$Spots))\n","tmp <- as.numeric(curated_df_2$Spots)\n","mean(tmp[tmp != 0])\n","median(tmp[tmp != 0])\n","\n","# spots: dataframe3\n","median(as.numeric(manual_dataframe$Spots))\n","tmp <- as.numeric(manual_dataframe$Spots)\n","mean(tmp[tmp != 0])\n","median(tmp[tmp != 0])\n","\n","# Bases: dataframe1\n","sum(is.na(curated_df$Bases))\n","min(as.numeric(curated_df$Bases))\n","length(which(curated_df$Bases == 0))\n","max(as.numeric(curated_df$Bases))\n","mean(as.numeric(curated_df$Bases))\n","sd(as.numeric(curated_df$Bases))\n","median(as.numeric(curated_df$Bases))\n","\n","tmp <- as.numeric(curated_df$Bases)\n","mean(tmp[tmp != 0])\n","median(tmp[tmp != 0])\n","\n","# Bases: dataframe2\n","median(as.numeric(curated_df_2$Bases))\n","tmp <- as.numeric(curated_df_2$Bases)\n","mean(tmp[tmp != 0])\n","median(tmp[tmp != 0])\n","\n","# Bases: dataframe3\n","median(as.numeric(manual_dataframe$Bases))\n","tmp <- as.numeric(manual_dataframe$Bases)\n","mean(tmp[tmp != 0])\n","median(tmp[tmp != 0])\n","\n","# Check AvgLength: dataframe1\n","length(which(curated_df$AvgLength == \"0\"))\n","tmp <- as.numeric(curated_df$AvgLength)\n","mean(tmp[tmp != 0])\n","median(tmp[tmp != 0])\n","\n","# Check AvgLength: dataframe2\n","length(which(curated_df_2$AvgLength == \"0\"))\n","tmp <- as.numeric(curated_df_2$AvgLength)\n","mean(tmp[tmp != 0])\n","median(tmp[tmp != 0])\n","\n","# Check AvgLength: dataframe3\n","length(which(manual_dataframe$AvgLength == \"0\"))\n","mean(as.numeric(manual_dataframe$AvgLength))\n","sd(as.numeric(manual_dataframe$AvgLength))\n","median(as.numeric(manual_dataframe$AvgLength))\n","\n","# insert size: dataframe1:\n","length(which(curated_df$Insert_Size == \"0\"))\n","mean(as.numeric(curated_df$Insert_Size))\n","sd(as.numeric(curated_df$Insert_Size))\n","median(as.numeric(curated_df$Insert_Size))\n","tmp <- as.numeric(curated_df$Insert_Size)\n","mean(tmp[tmp != 0])\n","median(tmp[tmp != 0])\n","# insert size: dataframe2:\n","length(which(curated_df_2$Insert_Size == \"0\"))\n","mean(as.numeric(curated_df_2$Insert_Size))\n","sd(as.numeric(curated_df_2$Insert_Size))\n","median(as.numeric(curated_df_2$Insert_Size))\n","tmp <- as.numeric(curated_df_2$Insert_Size)\n","mean(tmp[tmp != 0])\n","median(tmp[tmp != 0])\n","# insert size: dataframe3:\n","length(which(manual_dataframe$Insert_Size == \"0\"))\n","mean(as.numeric(manual_dataframe$Insert_Size))\n","sd(as.numeric(manual_dataframe$Insert_Size))\n","median(as.numeric(manual_dataframe$Insert_Size))\n","tmp <- as.numeric(manual_dataframe$Insert_Size)\n","mean(tmp[tmp != 0])\n","median(tmp[tmp != 0])\n","\n","# Sex:\n","unique(curated_df_2$Sex)\n","\n","#dataframe1\n","length(which(curated_df$Sex == \"male\"))\n","length(which(curated_df$Sex == \"female\"))\n","\n","#dataframe2\n","length(which(curated_df_2$Sex == \"male\"))\n","length(which(curated_df_2$Sex == \"female\"))\n","\n","dataframe3\n","length(which(manual_dataframe$Sex == \"male\"))\n","length(which(manual_dataframe$Sex == \"female\"))\n","\n","# Body site:\n","unique(curated_df_2$Body_Site)\n","\n","# dataframe1\n","length(which(curated_df$Body_Site == \"\"))\n","length(which(curated_df$Body_Site == \"Unknown\"))\n","length(which(curated_df$Body_Site == \"not collected\"))\n","length(which(curated_df$Body_Site == \"BODY_SITE\"))\n","length(which(curated_df$Body_Site == \"Unspecified\"))\n","length(which(curated_df$Body_Site == \"Not applicable\"))\n","length(which(curated_df$Body_Site == \"not applicable\"))\n","\n","# dataframe2\n","sum(is.na(curated_df_2$Body_Site))\n","length(which(curated_df_2$Body_Site == \"\"))\n","length(which(curated_df_2$Body_Site == \"Unknown\"))\n","length(which(curated_df_2$Body_Site == \"not collected\"))\n","length(which(curated_df_2$Body_Site == \"BODY_SITE\"))\n","length(which(curated_df_2$Body_Site == \"Unspecified\"))\n","length(which(curated_df_2$Body_Site == \"Not applicable\"))\n","length(which(curated_df_2$Body_Site == \"not applicable\"))\n","length(which(curated_df_2$Body_Site == \"control\"))\n","length(which(curated_df_2$Body_Site == \"ctrl\"))\n","length(which(curated_df_2$Body_Site == \"epidermis\"))\n","\n","# dataframe3\n","length(which(manual_dataframe$Body_Site == \"\"))\n","length(which(manual_dataframe$Body_Site == \"Unknown\"))\n","length(which(manual_dataframe$Body_Site == \"not collected\"))\n","length(which(manual_dataframe$Body_Site == \"BODY_SITE\"))\n","length(which(manual_dataframe$Body_Site == \"Unspecified\"))\n","length(which(manual_dataframe$Body_Site == \"Not applicable\"))\n","length(which(manual_dataframe$Body_Site == \"not applicable\"))\n","\n","# Journal\n","unique(manual_df$Journal)\n","unique(manual_df$WOS_Research_Areas)\n","unique(manual_df$WOS_Categories)\n","length(which(manual_df$Medicine_journal == \"Yes\"))"],"metadata":{"id":"Ry-9X42dfdqQ"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["## Code used to generate plots"],"metadata":{"id":"_eSBXsI2fh2a"}},{"cell_type":"code","source":["# Plots:\n","library(ggplot2)\n","\n","# Year of release:\n","\n","# data frame 2:\n","vec <- curated_df_2$Year_of_release\n","vec <- as.numeric(vec)\n","vec2 <- unique(vec)\n","count <- c()\n","\n","for (j in 1:length(vec2)){\n","  count <- append(count, length(which(vec==vec2[j])))\n","}\n","\n","#create a dataframe\n","plotdf <- data.frame(vec2, count)\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_line(stat=\"identity\") +\n","  scale_x_continuous(\"vec2\", labels = as.character(vec2), breaks = vec2) +\n","  xlab(\"Year of release\") + ylab(\"Number of samples\") +\n","  ggtitle(\"Year of release: samples (dataframe2)\") +\n","  theme_bw()\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_bar(stat=\"identity\") +\n","  scale_x_continuous(\"vec2\", labels = as.character(vec2), breaks = vec2) +\n","  xlab(\"Year of release\") + ylab(\"Number of samples\") +\n","  ggtitle(\"Year of release: samples (dataframe2\")\n","\n","# data frame 3:\n","\n","vec <- manual_dataframe$Year_of_release\n","vec <- as.numeric(vec)\n","vec2 <- unique(vec)\n","count <- c()\n","\n","for (j in 1:length(vec2)){\n","  count <- append(count, length(which(vec==vec2[j])))\n","}\n","\n","#create a dataframe\n","plotdf <- data.frame(vec2, count)\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_line(stat=\"identity\") +\n","  scale_x_continuous(\"vec2\", labels = as.character(vec2), breaks = vec2) +\n","  xlab(\"Year of release\") + ylab(\"Number of samples\") +\n","  ggtitle(\"Year of release: samples (dataframe3)\") +\n","  theme_bw()\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_bar(stat=\"identity\") +\n","  scale_x_continuous(\"vec2\", labels = as.character(vec2), breaks = vec2) +\n","  xlab(\"Year of release\") + ylab(\"Number of samples\") +\n","  ggtitle(\"Year of release: samples (dataframe3)\")\n","\n","\n","# data frame 3: (studies)\n","\n","vec <- manual_df$Year_Of_Publication\n","add <- unlist(strsplit(vec[68], \"[\\n]\"))\n","vec <- head(vec, -1)\n","vec <- append(vec, add)\n","rm(add)\n","vec <- as.numeric(vec)\n","vec2 <- unique(vec)\n","count <- c()\n","\n","for (j in 1:length(vec2)){\n","  count <- append(count, length(which(vec==vec2[j])))\n","}\n","\n","#create a dataframe\n","plotdf <- data.frame(vec2, count)\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_line(stat=\"identity\") +\n","  scale_x_continuous(\"vec2\", labels = as.character(vec2), breaks = vec2) +\n","  xlab(\"Year of release\") + ylab(\"Number of studies\") +  \n","  ggtitle(\"Year of release: studies (dataframe3)\") +\n","  theme_bw()\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_bar(stat=\"identity\") +\n","  scale_x_continuous(\"vec2\", labels = as.character(vec2), breaks = vec2) +\n","  xlab(\"Year of release\") + ylab(\"Number of studies\") +\n","  ggtitle(\"Year of release: studies (dataframe3)\")\n","\n","# data frame 1:\n","\n","vec <- curated_df$Year_of_release\n","vec <- as.numeric(vec)\n","vec2 <- unique(vec)\n","count <- c()\n","\n","for (j in 1:length(vec2)){\n","  count <- append(count, length(which(vec==vec2[j])))\n","}\n","\n","#create a dataframe\n","plotdf <- data.frame(vec2, count)\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_line(stat=\"identity\") +\n","  scale_x_continuous(\"vec2\", labels = as.character(vec2), breaks = vec2) +\n","  xlab(\"Year of release\") + ylab(\"Number of samples\") + \n","  ggtitle(\"Year of release: samples (dataframe1)\") +\n","  theme_bw()\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_bar(stat=\"identity\") +\n","  scale_x_continuous(\"vec2\", labels = as.character(vec2), breaks = vec2) +\n","  xlab(\"Year of release\") + ylab(\"Number of samples\") +\n","  ggtitle(\"Year of release: samples (dataframe1)\")\n","\n","\n","# Seq Platform info - daraframe2:\n","vec <- curated_df_2$Platform\n","vec2 <- unique(vec)\n","count <- c()\n","\n","for (j in 1:length(vec2)){\n","  count <- append(count, length(which(vec==vec2[j])))\n","}\n","\n","#create a dataframe\n","plotdf <- data.frame(vec2, count)\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_bar(stat=\"identity\") +\n","  labs(y= \"Samples\", x = \"Sequencing Platform\") +\n","  ggtitle(\"Seq Platform - daraframe2: (samples)\")\n","\n","\n","# Seq Platform info - daraframe3:\n","vec <- manual_dataframe$Platform\n","vec2 <- unique(vec)\n","count <- c()\n","\n","for (j in 1:length(vec2)){\n","  count <- append(count, length(which(vec==vec2[j])))\n","}\n","\n","#create a dataframe\n","plotdf <- data.frame(vec2, count)\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_bar(stat=\"identity\") +\n","  labs(y= \"Samples\", x = \"Sequencing Platform\") +\n","  ggtitle(\"Seq Platform - daraframe3: (samples)\")\n","\n","\n","# Seq Platform Model info - daraframe2:\n","vec <- curated_df_2$Model\n","vec2 <- unique(vec)\n","count <- c()\n","\n","for (j in 1:length(vec2)){\n","  count <- append(count, length(which(vec==vec2[j])))\n","}\n","\n","#create a dataframe\n","plotdf <- data.frame(vec2, count)\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_bar(stat=\"identity\") +\n","  labs(y= \"Samples\", x = \"Sequencing Platform Model\") +\n","  ggtitle(\"Seq Platform Model info - daraframe2: (samples)\")\n","\n","\n","# Seq Platform Model info - daraframe3:\n","vec <- manual_dataframe$Model\n","vec2 <- unique(vec)\n","count <- c()\n","\n","for (j in 1:length(vec2)){\n","  count <- append(count, length(which(vec==vec2[j])))\n","}\n","\n","#create a dataframe\n","plotdf <- data.frame(vec2, count)\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_bar(stat=\"identity\") +\n","  labs(y= \"Samples\", x = \"Sequencing Platform Model\")  +\n","  ggtitle(\"Seq Platform Model info - daraframe3: (samples)\")\n","\n","\n","# Seq Platform Model info - daraframe3: (studies)\n","vec <- manual_df$Seq_Platform\n","vec2 <- unique(vec)\n","count <- c()\n","\n","for (j in 1:length(vec2)){\n","  count <- append(count, length(which(vec==vec2[j])))\n","}\n","\n","#create a dataframe\n","plotdf <- data.frame(vec2, count)\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_bar(stat=\"identity\") +\n","  theme(axis.text.x = element_text(angle = 45, hjust=1)) +\n","  labs(y= \"Studies\", x = \"Sequencing Platform Model\") +\n","  ggtitle(\"Seq Platform Model info - daraframe3: (studies)\")\n","\n","\n","# Info on the 16S region: (studies) - dataset 3\n","vec <- manual_df$vRegion\n","vec2 <- unique(vec)\n","count <- c()\n","\n","for (j in 1:length(vec2)){\n","  count <- append(count, length(which(vec==vec2[j])))\n","}\n","\n","#create a dataframe\n","plotdf <- data.frame(vec2, count)\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_bar(stat=\"identity\") +\n","  theme(axis.text.x = element_text(angle = 45, hjust=1)) +\n","  labs(y= \"Studies\", x = \"16S rRNA hypervariable region sequenced\") +\n","  ggtitle(\"16S rRNA region - daraframe3: (studies)\")\n","\n","# Info on the 16S region: (samples) - dataset 3\n","vec <- manual_dataframe$Region_16S\n","vec2 <- unique(vec)\n","count <- c()\n","\n","for (j in 1:length(vec2)){\n","  count <- append(count, length(which(vec==vec2[j])))\n","}\n","\n","#create a dataframe\n","plotdf <- data.frame(vec2, count)\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_bar(stat=\"identity\") +\n","  theme(axis.text.x = element_text(angle = 45, hjust=1)) +\n","  labs(y= \"Samples\", x = \"16S rRNA hypervariable region sequenced\") +\n","  ggtitle(\"16S rRNA region - daraframe3: (samples)\")\n","\n","\n","# Collection method (for the samples): -dataframe3\n","vec <- manual_dataframe$Collection_method\n","vec2 <- unique(vec)\n","count <- c()\n","\n","for (j in 1:length(vec2)){\n","  count <- append(count, length(which(vec==vec2[j])))\n","}\n","\n","#create a dataframe\n","plotdf <- data.frame(vec2, count)\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_bar(stat=\"identity\") +\n","  theme(axis.text.x = element_text(angle = 45, hjust=1)) +\n","  labs(y= \"Samples\", x = \"Collection method\") +\n","  ggtitle(\"Collection method - daraframe3: (samples)\")\n","\n","# Collection method (for the studies):-dataframe3\n","vec <- manual_df$collection_method\n","vec2 <- unique(vec)\n","count <- c()\n","\n","for (j in 1:length(vec2)){\n","  count <- append(count, length(which(vec==vec2[j])))\n","}\n","\n","#create a dataframe\n","plotdf <- data.frame(vec2, count)\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_bar(stat=\"identity\") +\n","  theme(axis.text.x = element_text(angle = 45, hjust=1)) +\n","  labs(y= \"Studies\", x = \"Collection method\") +\n","  ggtitle(\"Collection method - daraframe3: (studies)\")\n","\n","# Clustering method used in the studies:\n","vec <- manual_df$clustering_used\n","vec2 <- unique(vec)\n","count <- c()\n","\n","for (j in 1:length(vec2)){\n","  count <- append(count, length(which(vec==vec2[j])))\n","}\n","\n","#create a dataframe\n","plotdf <- data.frame(vec2, count)\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_bar(stat=\"identity\") +\n","  theme(axis.text.x = element_text(angle = 45, hjust=1)) +\n","  labs(y= \"Studies\", x = \"Clustering method\") +\n","  ggtitle(\"Clustering method - daraframe3: (studies)\")\n","\n","# Disease and conditions: (studies) - dataframe3\n","vec <- manual_df$disease.condition\n","vec2 <- unique(vec)\n","count <- c()\n","\n","for (j in 1:length(vec2)){\n","  count <- append(count, length(which(vec==vec2[j])))\n","}\n","\n","#create a dataframe\n","plotdf <- data.frame(vec2, count)\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_bar(stat=\"identity\") +\n","  theme(axis.text.x = element_text(angle = 45, hjust=1)) +\n","  labs(y= \"Studies\", x = \"Disease/condition\") +\n","  ggtitle(\"Diseases and conditions - daraframe3: (studies)\")\n","\n","\n","# Taxonomic database:\n","vec <- manual_df$taxon_database.used_in_the_study.\n","vec2 <- unique(vec)\n","count <- c()\n","\n","for (j in 1:length(vec2)){\n","  count <- append(count, length(which(vec==vec2[j])))\n","}\n","\n","#create a dataframe\n","plotdf <- data.frame(vec2, count)\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_bar(stat=\"identity\") +\n","  theme(axis.text.x = element_text(angle = 45, hjust=1)) +\n","  labs(y= \"Studies\", x = \"Taxonomic database used in the study\") +\n","  ggtitle(\"Taxonomic database used in the study - daraframe3\")\n","\n","\n","# Location of the study:\n","vec <- manual_df$Location\n","vec2 <- unique(vec)\n","count <- c()\n","\n","for (j in 1:length(vec2)){\n","  count <- append(count, length(which(vec==vec2[j])))\n","}\n","\n","#create a dataframe\n","plotdf <- data.frame(vec2, count)\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_bar(stat=\"identity\") +\n","  theme(axis.text.x = element_text(angle = 45, hjust=1)) +\n","  labs(y= \"Studies\", x = \"Location of the study\") +\n","  ggtitle(\"Location of the study - daraframe3\")\n","\n","names_location <- plotdf$vec2\n","number <- c()\n","for(k in 1:length(names_location)){\n","  number <- append(number, length(which(manual_df$Location == names_location[k])))\n","}\n","plotdf <- data.frame(names_location, number)\n","\n","library(viridis)\n","ggplot(plotdf, aes(x = names_location, y = names_location, size = number, fill = number)) +\n","  geom_point(shape = 21) +\n","  scale_fill_viridis_c(option = \"plasma\")\n","\n","\n","# Sex:\n","\n","vec2 <- c(\"female_df1\", \"male_df1\", \"NA_df1\", \"female_df2\", \"male_df2\", \"NA_df2\", \"female_df3\", \"male_df3\", \"NA_df3\")\n","count <- c()\n","\n","count <- append(count, length(which(curated_df$Sex == \"male\")))\n","n <- length(which(curated_df$Sex == \"male\"))\n","count <- append(count, length(which(curated_df$Sex == \"female\")))\n","n <- n + length(which(curated_df$Sex == \"female\"))\n","l <- nrow(curated_df)\n","diff <- l-n\n","count <- append(count, diff)\n","\n","count <- append(count, length(which(curated_df_2$Sex == \"male\")))\n","n <- length(which(curated_df_2$Sex == \"male\"))\n","count <- append(count, length(which(curated_df_2$Sex == \"female\")))\n","n <- n + length(which(curated_df_2$Sex == \"female\"))\n","l <- nrow(curated_df_2)\n","diff <- l-n\n","count <- append(count, diff)\n","\n","count <- append(count, length(which(manual_dataframe$Sex == \"male\")))\n","n <- length(which(manual_dataframe$Sex == \"male\"))\n","count <- append(count, length(which(manual_dataframe$Sex == \"female\")))\n","n <- n + length(which(manual_dataframe$Sex == \"female\"))\n","l <- nrow(manual_dataframe)\n","diff <- l-n\n","count <- append(count, diff)\n","\n","#create a dataframe\n","plotdf <- data.frame(vec2, count)\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_bar(stat=\"identity\") +\n","  theme(axis.text.x = element_text(angle = 45, hjust=1)) +\n","  labs(y= \"Samples\", x = \"Sex distribution among datasets\") +\n","  ggtitle(\"Sex distribution among datasets\")\n","\n","\n","# Taxon ID:\n","\n","vec2 <- c(\"df1\", \"df2\", \"df3\")\n","count <- c()\n","count <- append(count, length(unique(curated_df$TaxID)))\n","count <- append(count, length(unique(curated_df_2$TaxID)))\n","count <- append(count, length(unique(manual_dataframe$TaxID)))\n","\n","#create a dataframe\n","plotdf <- data.frame(vec2, count)\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_bar(stat=\"identity\") +\n","  xlab(\"Data Frames\") + ylab(\"Number of Taxon ID\") +\n","  ggtitle(\"TaxID\") +\n","  theme_bw()\n","\n","\n","# taxID\n","\n","vec <- manual_dataframe$Scientific_Name\n","vec2 <- unique(vec)\n","count <- c()\n","\n","for (j in 1:length(vec2)){\n","  count <- append(count, length(which(vec==vec2[j])))\n","}\n","\n","# use log \n","count <- log(count)\n","\n","#create a dataframe\n","plotdf <- data.frame(vec2, count)\n","\n","\n","ggplot(plotdf, aes(x=\"\", y=count, fill=vec2)) +\n","  geom_bar(stat=\"identity\", width=1) + \n","  scale_fill_manual(values = c(\"dodgerblue2\", \n","                               \"#E31A1C\", \n","                               \"green4\",\n","                               \"#6A3D9A\",\n","                               \"#FF7F00\",\n","                               \"gold1\",\n","                               \"skyblue2\", \n","                               \"#FB9A99\",\n","                               \"palegreen2\",\n","                               \"#CAB2D6\",\n","                               \"#FDBF6F\",\n","                               \"gray70\", \n","                               \"khaki2\",\n","                               \"maroon\", \n","                               \"orchid1\", \n","                               \"deeppink1\", \n","                               \"blue1\", \n","                               \"steelblue4\",\n","                               \"darkturquoise\", \n","                               \"green1\", \n","                               \"yellow4\", \n","                               \"yellow3\",\n","                               \"darkorange4\", \n","                               \"brown\")) +\n"," coord_polar(\"y\", start=0)\n","\n","\n","# Spots Plot\n","\n","vec2 <- c(\"df1\", \"df2\", \"df3\")\n","count <- c()\n","# Spots: dataframe1\n","tmp <- as.numeric(curated_df$Spots)\n","count <- append(count, median(tmp[tmp != 0])) \n","# spots: dataframe2\n","tmp <- as.numeric(curated_df_2$Spots)\n","count <- append(count, median(tmp[tmp != 0])) \n","# spots: dataframe3\n","tmp <- as.numeric(manual_dataframe$Spots)\n","count <- append(count, median(tmp[tmp != 0])) \n","\n","#create a dataframe\n","plotdf <- data.frame(vec2, count)\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_bar(stat=\"identity\") +\n","  xlab(\"Data Frames\") + ylab(\"Spots\") +\n","  ggtitle(\"Spots\") +\n","  theme_bw()\n","\n","\n","# Bases Plot\n","\n","vec2 <- c(\"df1\", \"df2\", \"df3\")\n","count <- c()\n","# Spots: dataframe1\n","tmp <- as.numeric(curated_df$Bases)\n","count <- append(count, median(tmp[tmp != 0])) \n","# spots: dataframe2\n","tmp <- as.numeric(curated_df_2$Bases)\n","count <- append(count, median(tmp[tmp != 0])) \n","# spots: dataframe3\n","tmp <- as.numeric(manual_dataframe$Bases)\n","count <- append(count, median(tmp[tmp != 0])) \n","\n","#create a dataframe\n","plotdf <- data.frame(vec2, count)\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_bar(stat=\"identity\") +\n","  xlab(\"Data Frames\") + ylab(\"Bases\") +\n","  ggtitle(\"Bases\") +\n","  theme_bw()\n","\n","\n","# Avg Length Plot\n","\n","vec2 <- c(\"df1\", \"df2\", \"df3\")\n","count <- c()\n","# Spots: dataframe1\n","tmp <- as.numeric(curated_df$AvgLength)\n","count <- append(count, median(tmp[tmp != 0])) \n","# spots: dataframe2\n","tmp <- as.numeric(curated_df_2$AvgLength)\n","count <- append(count, median(tmp[tmp != 0])) \n","# spots: dataframe3\n","tmp <- as.numeric(manual_dataframe$AvgLength)\n","count <- append(count, median(tmp[tmp != 0])) \n","\n","#create a dataframe\n","plotdf <- data.frame(vec2, count)\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_bar(stat=\"identity\") +\n","  xlab(\"Data Frames\") + ylab(\"Average Length\") +\n","  ggtitle(\"Average Length\") +\n","  theme_bw()\n","\n","# Avg Length - df1\n","vec <- curated_df$AvgLength\n","vec <- as.numeric(vec)\n","vec2 <- unique(vec)\n","count <- c()\n","\n","for (j in 1:length(vec2)){\n","  count <- append(count, length(which(vec==vec2[j])))\n","}\n","\n","#create a dataframe\n","plotdf <- data.frame(vec2, count)\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_bar(stat=\"identity\") +\n","  scale_x_binned() +\n","  xlab(\"Average Length\") + ylab(\"Number of samples\") +\n","  ggtitle(\"Average Length (dataframe1)\") +\n","  theme(axis.text.x = element_text(angle = 45, hjust=1))\n","\n","# Avg Length - df2\n","vec <- curated_df_2$AvgLength\n","vec <- as.numeric(vec)\n","vec2 <- unique(vec)\n","count <- c()\n","\n","for (j in 1:length(vec2)){\n","  count <- append(count, length(which(vec==vec2[j])))\n","}\n","\n","#create a dataframe\n","plotdf <- data.frame(vec2, count)\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_bar(stat=\"identity\") +\n","  scale_x_binned() +\n","  xlab(\"Average Length\") + ylab(\"Number of samples\") +\n","  ggtitle(\"Average Length (dataframe2)\") +\n","  theme(axis.text.x = element_text(angle = 45, hjust=1))\n","\n","# Avg Length - df3\n","vec <- manual_dataframe$AvgLength\n","vec <- as.numeric(vec)\n","vec2 <- unique(vec)\n","count <- c()\n","\n","for (j in 1:length(vec2)){\n","  count <- append(count, length(which(vec==vec2[j])))\n","}\n","\n","#create a dataframe\n","plotdf <- data.frame(vec2, count)\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_bar(stat=\"identity\") +\n","  scale_x_binned() +\n","  xlab(\"Average Length\") + ylab(\"Number of samples\") +\n","  ggtitle(\"Average Length (dataframe3)\") +\n","  theme(axis.text.x = element_text(angle = 45, hjust=1))\n","\n","\n","# Insert size Plot\n","\n","vec2 <- c(\"df1\", \"df2\", \"df3\")\n","count <- c()\n","# Spots: dataframe1\n","tmp <- as.numeric(curated_df$Insert_Size)\n","count <- append(count, median(tmp[tmp != 0])) \n","# spots: dataframe2\n","tmp <- as.numeric(curated_df_2$Insert_Size)\n","count <- append(count, median(tmp[tmp != 0])) \n","# spots: dataframe3\n","tmp <- as.numeric(manual_dataframe$Insert_Size)\n","count <- append(count, median(tmp[tmp != 0])) \n","\n","#create a dataframe\n","plotdf <- data.frame(vec2, count)\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_bar(stat=\"identity\") +\n","  xlab(\"Data Frames\") + ylab(\"Insert Size\") +\n","  ggtitle(\"Insert Size\") +\n","  theme_bw()\n","\n","# Insert Size - df1\n","vec <- curated_df$Insert_Size\n","vec <- as.numeric(vec)\n","vec2 <- unique(vec)\n","vec2 <- vec2[2:length(vec2)] # remove 0\n","count <- c()\n","\n","for (j in 1:length(vec2)){\n","  count <- append(count, length(which(vec==vec2[j])))\n","}\n","\n","#create a dataframe\n","plotdf <- data.frame(vec2, count)\n","plotdf <- plotdf %>% filter(vec2 < 1000)\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_bar(stat=\"identity\") +\n","  scale_x_binned(n.breaks = 10) +\n","  xlab(\"Insert Size\") + ylab(\"Number of samples\") +\n","  ggtitle(\"Insert Size (dataframe1)\") +\n","  theme(axis.text.x = element_text(angle = 45, hjust=1))\n","\n","# Insert Size - df2\n","vec <- curated_df_2$Insert_Size\n","vec <- as.numeric(vec)\n","vec2 <- unique(vec)\n","vec2 <- vec2[2:length(vec2)] # remove 0\n","count <- c()\n","\n","for (j in 1:length(vec2)){\n","  count <- append(count, length(which(vec==vec2[j])))\n","}\n","\n","#create a dataframe\n","plotdf <- data.frame(vec2, count)\n","plotdf <- plotdf %>% filter(vec2 < 1000)\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_bar(stat=\"identity\") +\n","  scale_x_binned(n.breaks = 10) +\n","  xlab(\"Insert Size\") + ylab(\"Number of samples\") +\n","  ggtitle(\"Insert Size (dataframe2)\") +\n","  theme(axis.text.x = element_text(angle = 45, hjust=1))\n","\n","# Insert Size - df3\n","vec <- manual_dataframe$Insert_Size\n","vec <- as.numeric(vec)\n","vec2 <- unique(vec)\n","vec2 <- vec2[2:length(vec2)] # remove 0\n","count <- c()\n","\n","for (j in 1:length(vec2)){\n","  count <- append(count, length(which(vec==vec2[j])))\n","}\n","\n","#create a dataframe\n","plotdf <- data.frame(vec2, count)\n","plotdf <- plotdf %>% filter(vec2 < 1000)\n","\n","ggplot(data=plotdf, aes(x=vec2, y=count)) +\n","  geom_bar(stat=\"identity\") +\n","  scale_x_binned(n.breaks = 10) +\n","  xlab(\"Insert Size\") + ylab(\"Number of samples\") +\n","  ggtitle(\"Insert Size (dataframe3)\") +\n","  theme(axis.text.x = element_text(angle = 45, hjust=1))"],"metadata":{"id":"TWS_Xn2jflSL"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["### Contacts\n","Name: Giulia Agostinetto\n","\n","e-mail: g.agostinetto@campus.unimib.it\n","\n","Name: Davide Bozzi\n","\n","e-mail: d.bozzi1@campus.unimib.it\n","\n","Please refer to both contacts for further information or issues related to the framework."],"metadata":{"id":"rH5G-Tc_foPM"}}]}